test/unit/lib/plugins/aws/package/compile/events/schedule.test.js

```diff
@@ -1,295 +1,323 @@
+
+      scheduleCfResources.push(scheduleCfResource);
+    }
+  }
+
+  return scheduleCfResources;
+}
+
+describe('ScheduleEvents', () => {
+  it('should create corresponding resources when schedule events are given', async () => {
+    const events = [
+      {
+        schedule: {
+          rate: 'rate(10 minutes)',
+          enabled: false,
         },
-      };
-
-      awsCompileScheduledEvents.compileScheduledEvents();
-
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstEventsRuleSchedule1.Properties.State
-      ).to.equal('DISABLED');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstEventsRuleSchedule2.Properties.State
-      ).to.equal('ENABLED');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstEventsRuleSchedule3.Properties.State
-      ).to.equal('ENABLED');
-      expect(
-        awsCompileScheduledEvents.serverless.service.provider.compiledCloudFormationTemplate
-          .Resources.FirstEventsRuleSchedule4.Properties.State
-      ).to.equal('ENABLED');
-    });
+      },
+      {
+        schedule: {
+          rate: 'rate(10 minutes)',
+          enabled: true,
+        },
+      },
+      {
+        schedule: 'rate(10 minutes)',
+      },
+      {
+        schedule: 'cron(5,35 12 ? * 6l 2002-2005)',
+      },
+      {
+        schedule: {
+          rate: ['cron(0 0/4 ? * MON-FRI *)', 'rate(1 hour)'],
+        },
+      },
+    ];
 
-    it('should respect name variable', () => {
-      awsCompileScheduledEvents.serverless.service.functions = {
-        first: {
-          events: [
-            {
-              schedule: {
-                rate: 'rate(10 minutes)',
-                enabled: false,
-                name: 'your-scheduled-event-name',
-              },
-            },
-          ],
+    const scheduleCfResources = await run(events);
+
+    for (let i = 0; i < events.length; i += 1) {
+      expect(scheduleCfResources[i].Type).to.equal('AWS::Events::Rule');
+    }
+  });
+
+  it('should respect enabled variable, defaulting to true', async () => {
+    const events = [
+      {
+        schedule: {
+          rate: 'rate(10 minutes)',
+          enabled: false,
+        },
+      },
+      {
+        schedule: {
+          rate: 'rate(10 minutes)',
+          enabled: true,
+        },
+      },
+      {
+        schedule: {
+          rate: 'rate(10 minutes)',
+        },
+      },
+      {
+        schedule: 'rate(10 minutes)',
+      },
+      {
+        schedule: {
+          rate: ['cron(0 0/4 ? * MON-FRI *)', 'rate(1 hour)'],
+          enabled: false,
         },
-      };
+      },
+    ];
 
-      awsCompileScheduledEvents.compileScheduledEvents();
+    const scheduleCfResources = await run(events);
```

Comment:
Instead of having separate `run`s, do you think we could configure all these cases under a single `runServerless` that is executed in `before` hook and run only assertions in all other tests? The reason for this approach is to limit `runServerless` runs as they're costly/lengthy. 

Of course the tests for errors should still be separate.

Reply:
Oh, I see. Will do.

Reply:
:heavy_check_mark: